### PR TITLE
Added automation for flyteidl proto

### PR DESCRIPTION
**Pull request by rajatk-31**
# TL;DR
- Added a new make target `update_flyteidl` .
- Added a shell script for bump new flyteidl proto.
- Added new Github workflow for automation

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/1384

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/1384_

### CODE REVIEW

File: scripts/update_flyteidl.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+trap 'rm -fr $OUT' EXIT
+
+
+FLYTEIDL_VERSION=$(curl --silent "https://api.github.com/repos/flyteorg/flyteidl/releases/latest" | jq -r .tag_name)
+git clone https://github.com/flyteorg/flyteidl.git "${OUT}" --branch "${FLYTEIDL_VERSION}"
+
+mv ${OUT}/protos/flyteidl/admin/*  jflyte/src/main/proto/flyteidl/admin/
+mv ${OUT}/protos/flyteidl/core/*  jflyte/src/main/proto/flyteidl/core/
+mv ${OUT}/protos/flyteidl/event/*  jflyte/src/main/proto/flyteidl/event/
+mv ${OUT}/protos/flyteidl/service/*  jflyte/src/main/proto/flyteidl/service/
```
Comment (evalsocket) [inline scripts/update_flyteidl.sh:22]: It would be great if we can use a loop for running `mv` command. Use an array to declare all the services

File: .github/workflows/flyteidl-automation.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: Update flyteidl Automation
+on:
```
Comment (evalsocket) [inline .github/workflows/flyteidl-automation.yml:2]: Is it tested ?

Comment (rajatk-31) [reply .github/workflows/flyteidl-automation.yml:2]: Yes.
Workflow URL: https://github.com/rajatk-31/flytekit-java/runs/3461746397?check_suite_focus=true

Comment (rajatk-31) [reply .github/workflows/flyteidl-automation.yml:2]: PR: https://github.com/rajatk-31/flytekit-java/pull/1

**Review by evalsocket — APPROVED** [review]

File: .github/workflows/flyteidl-automation.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  workflow_dispatch:
+
+jobs:
+  update-boilerplate:
+    name: Update flyteidl
```
Comment (ckiosidis) [inline .github/workflows/flyteidl-automation.yml:7]: the job is called `update-boilerplate` but the name is `Update flyteidl`
Also in other places it's called boilerplate, should we keep the same naming convention? https://github.com/flyteorg/flyteconsole/blob/master/.github/workflows/boilerplate-automation.yml

Comment (ckiosidis) [reply .github/workflows/flyteidl-automation.yml:7]: ah now i see that they are doing different things.
in the other projects flyteidls are not shared in the same way.

File: scripts/update_flyteidl.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+# WARNING: THIS FILE IS MANAGED IN THE 'BOILERPLATE' REPO AND COPIED TO OTHER REPOSITORIES.
+# ONLY EDIT THIS FILE FROM WITHIN THE 'FLYTEORG/BOILERPLATE' REPOSITORY:
+# 
+# TO OPT OUT OF UPDATES, SEE https://github.com/flyteorg/boilerplate/blob/master/Readme.rst
```
Comment (ckiosidis) [inline scripts/update_flyteidl.sh:6]: I couldnt find the original file that is mentioned here, could you please add a link?

Comment (rajatk-31) [reply scripts/update_flyteidl.sh:6]: Yes , here is the link : https://github.com/rajatk-31/flytekit-java/blob/master/scripts/update_flyteidl.sh

Comment (ckiosidis) [reply scripts/update_flyteidl.sh:6]: Thanks, I actually meant a link towards the boilerplate repo. The warning suggests that the file exists there. Does it?

Comment (rajatk-31) [reply scripts/update_flyteidl.sh:6]: https://github.com/rajatk-31/flytekit-java/blob/master/boilerplate/update.sh

Comment (ckiosidis) [reply scripts/update_flyteidl.sh:6]: should this be included in https://github.com/flyteorg/boilerplate/ ?

Comment (evalsocket) [reply scripts/update_flyteidl.sh:6]: I think no, It's very specific to flytekit-java

Comment (evalsocket) [reply scripts/update_flyteidl.sh:6]: @rajatk-31 Please remove the boilerplate comment from script

File: .github/workflows/flyteidl-automation.yml (hunk 3 above)
Comment (evalsocket) [reply .github/workflows/flyteidl-automation.yml:7]: for now, let's copy the flyteidl and you will take care of the generatation part...In meanwhile I will release the maven package in flyteidl. It would be great if they are available in maven

After this, you can decide what to do next in flytekit-java

**Review by ckiosidis — APPROVED** [review]

File: scripts/update_flyteidl.sh [hunk 5]
```diff
@@ -0,0 +1,17 @@
+#copying latest proto from flyteidl to flytekit-java
```
Comment (narape) [inline scripts/update_flyteidl.sh:1]: @evalsocket this file needs execution permission for the make to work

**Review by narape — CHANGES_REQUESTED** [review]

File: scripts/update_flyteidl.sh (hunk 5 above)
Comment (evalsocket) [reply scripts/update_flyteidl.sh:1]: yes....@rajatk-31 did you added the right permission to `update_flyteidl.sh`
